install_config/router/customized_haproxy_router.adoc

```diff
@@ -115,21 +115,24 @@ The *define* action names the file that will contain the processed template.
 [cols="2*", options="header"]
 |===
 |Function | Meaning
-|`*endpointsForAlias(alias ServiceAliasConfig, svc ServiceUnit) []Endpoint*` | Returns the list of valid endpoints.
-|`*env(variable, default string) string*` | Tries to get the named environment variable from the pod. Returns the second
-argument if the variable cannot be read or is empty.
+|`*processEndpointsForAlias(alias ServiceAliasConfig, svc ServiceUnit, action string) []Endpoint*` |
+Returns the list of valid endpoints. When action is "shuffle", the order of endpoints is randomized.
+|`*env(variable, default ...string) string*` | Tries to get the named environment variable from the pod.
+If it is not defined or empty, it returns the optional second argument.
 |`*matchPattern(pattern, s string) bool*` | The first argument is a string that contains the regular
-expression, the second argument is the variable to test. Returns a Boolean value indicating whether the regular
-expression provided as the first argument matches the string provided as the second argument.
+expression, the second argument is the variable to test. Returns a Boolean value indicating whether
+the regular expression provided as the first argument matches the string provided as the second argument.
 |`*isInteger(s string) bool*` | Determines if a given variable is an integer.
-|`*matchValues(s string, allowedValues ...string) bool*` | Compares a given string to a list of allowed strings.
-|`*genSubdomainWildcardRegexp(hostname, path string, exactPath bool) string*` | Generates a regular expression matching the
-subdomain for hosts (and paths) with a wildcard policy.
-|`*generateRouteRegexp(hostname, path string, wildcard bool) string*` | Generates a regular expression matching the route
-hosts (and paths). The first argument is the host name, the second is the path,
+|`*firstMatch(s string, allowedValues ...string) bool*` | Compares a given string to a list of allowed
```

Comment:
The first argument of `firstMatch` is a regular expression.

Reply:
changed